### ISSUE TITLE
panic on parse-duration when no duration given

### ISSUE THREAD

**Issue by The-King-of-Toasters** [natsukagami/mpd-mpris#2]
I recently wanted to listen to some SID (C64 Music) files and downloaded the whole library [here](https://www.hvsc.c64.org/#download). MPD can decode them by compiling with `--enable-sidplay` with the `liblibsidplayfp` package (on Arch Linux), and enabling the decoder in the conf:
```
decoder {
	plugin			"sidplay"
	songlength_database	"/home/stephen/Music/C64Music/DOCUMENTS/Songlengths.txt"
	default_songlength	"120"
	filter 			"true"
}
```

When running mpd-mpris when playing a sid file, the following error occurs:
```
panic: Parse duration: strconv.ParseFloat: parsing "": invalid syntax

goroutine 1 [running]:
github.com/natsukagami/mpd-mpris.(*Player).properties(0xc420098120, 0xc42014e000)
        /tmp/yaytmp-1000/mpd-mpris/src/build/src/github.com/natsukagami/mpd-mpris/player.go:208 +0x14be
github.com/natsukagami/mpd-mpris.NewInstance(0xc4200a41e0, 0x3, 0xc420094240, 0xe)
        /tmp/yaytmp-1000/mpd-mpris/src/build/src/github.com/natsukagami/mpd-mpris/instance.go:50 +0x25c
main.main()
        /tmp/yaytmp-1000/mpd-mpris/src/build/src/github.com/natsukagami/mpd-mpris/cmd/mpd-mpris/main.go:43 +0x14e
```

When playing stops and mpd-mpris is run, this error occurs:
```
2018/08/18 18:49:01 strconv.Atoi: parsing "": invalid syntax
github.com/natsukagami/mpd-mpris/mpd.StatusFromAttrs
        /tmp/yaytmp-1000/mpd-mpris/src/build/src/github.com/natsukagami/mpd-mpris/mpd/status.go:40
github.com/natsukagami/mpd-mpris/mpd.(*Client).Status
        /tmp/yaytmp-1000/mpd-mpris/src/build/src/github.com/natsukagami/mpd-mpris/mpd/client.go:156
github.com/natsukagami/mpd-mpris.(*Player).properties
        /tmp/yaytmp-1000/mpd-mpris/src/build/src/github.com/natsukagami/mpd-mpris/player.go:183
github.com/natsukagami/mpd-mpris.NewInstance
        /tmp/yaytmp-1000/mpd-mpris/src/build/src/github.com/natsukagami/mpd-mpris/instance.go:50
main.main
        /tmp/yaytmp-1000/mpd-mpris/src/build/src/github.com/natsukagami/mpd-mpris/cmd/mpd-mpris/main.go:43
runtime.main
        /usr/lib/go/src/runtime/proc.go:198
runtime.goexit
        /usr/lib/go/src/runtime/asm_amd64.s:2361
```

When switching to a SID file, this error repeats:
```
2018/08/18 18:44:09 Parse duration: strconv.ParseFloat: parsing "": invalid syntax
```

I believe this is due to SID files not having a built in length, rather mpd parses the lengths from Songlengths.txt and prints "-:--" as its time.

**Comment by natsukagami** [comment]
I see, let me work on it.

**Comment by natsukagami** [comment]
Clearly, I did not expect mpd would play files with no length information!

Please try running on the latest commit (834e23150419e197f0571ef8be7c288d8838797f) to see if the problem has been fixed!

**Comment by The-King-of-Toasters** [comment]
It words, but now when the playlist is clear or playback is over it spits out
```
2018/08/18 18:44:09 Parse duration: strconv.ParseFloat: parsing "": invalid syntax
```
over and over.

Starting it from the cleared state gives a panic:
```
2018/08/21 01:32:52 strconv.Atoi: parsing "": invalid syntax
github.com/natsukagami/mpd-mpris/mpd.StatusFromAttrs
        /home/stephen/go/src/github.com/natsukagami/mpd-mpris/mpd/status.go:43
github.com/natsukagami/mpd-mpris/mpd.(*Client).Status
        /home/stephen/go/src/github.com/natsukagami/mpd-mpris/mpd/client.go:156
github.com/natsukagami/mpd-mpris.(*Player).properties
        /home/stephen/go/src/github.com/natsukagami/mpd-mpris/player.go:186
github.com/natsukagami/mpd-mpris.NewInstance
        /home/stephen/go/src/github.com/natsukagami/mpd-mpris/instance.go:50
main.main
        /home/stephen/go/src/github.com/natsukagami/mpd-mpris/cmd/mpd-mpris/main.go:43
runtime.main
        /usr/lib/go/src/runtime/proc.go:198
runtime.goexit
        /usr/lib/go/src/runtime/asm_amd64.s:2361
```

**Comment by natsukagami** [comment]
I guess it is better to just ignore the parse errors, although they are weird!

I ruled the empty string case out yet the error still happens. I guess it is some invalid characters not showing up?

**Comment by JakeStanger** [comment]
I'm getting the `strconv.Atoi: parsing "": invalid syntax` error on a paused state with an MP3 track. This is before I've unpaused MPD at any point since it started up.

**Comment by natsukagami** [comment]
Hi! May I get your mp3's metadata?

**Comment by JakeStanger** [comment]
This seems to be true for any track on further testing so I don't think metadata will help. If the player is in a paused state when mpd-mpris starts, it won't work.

**Comment by natsukagami** [comment]
I see. Let me see what I can do

**Comment by natsukagami** [comment]
It seems that there's some problem regarding the mpd library I am using or my wrapper around it. 

Unfortunately, it is currently exam season for me now, I cannot focus on this. It is on top of my to-do list after examinations now

**Comment by JakeStanger** [comment]
What sort of timescale are we looking at? I might be able to look into it and have a play around.

**Comment by natsukagami** [comment]
@JakeStanger sorry for the late reply. It seems that sometimes mpd status in `paused` state would not send information about volumes. 
I took `volume` to be a must-have value, hence the parsing panic. I will work on a fix, and this time I will be reading the mpd specs to make sure I do things correctly.

**Comment by natsukagami** [comment]
@The-King-of-Toasters @JakeStanger Please give the latest changes on `master` a try.

**Comment by dpgeekzero** [comment]
@natsukagami Not the original poster, but the changes are working great for me.  I stream alot of SomaFM and a lot of the tracks don't provide length information which was causing the current release to crash for me.

**Comment by natsukagami** [comment]
Cool! I might pack a release tomorrow.

**Comment by natsukagami** [comment]
Closing this since it seems to be solved in 0.2.1

**Comment by JakeStanger** [comment]
Sorry I didn't have time to test, thanks for the update!
